CMakeLists.txt

```diff
@@ -473,6 +473,14 @@ else()
   set(SWIFT_CROSS_COMPILING TRUE)
 endif()
 
+# Check if a prebuilt clang path was passed in, as this variable will be
+# assigned if not, in SwiftSharedCMakeConfig.
+if("${SWIFT_NATIVE_CLANG_TOOLS_PATH}" STREQUAL "")
+  set(SWIFT_PREBUILT_CLANG FALSE)
+else()
+  set(SWIFT_PREBUILT_CLANG TRUE)
+endif()
```

Comment:
Is there a reason to prefer this over just using standard CMAKE_C{,XX}_COMPILER?

Reply:
Yes, because the compiler and stdlib are built using potentially three clang compilers: the `--host-cc` is used to build a native Swift compiler, while the stdlib can also be built with either a freshly built Swift-forked clang that is usually built alongside this repo or a prebuilt clang that is passed in through `--native-clang-tools-path`. This new variable and `SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER` determine which of those three possible clangs to use when building the C++ portions of the stdlib.